File: web/src/main/java/com/github/dkurata38/open_web_library/web/infrastructure/library/AddedLibraryItemRepositoryImpl.java
package com.github.dkurata38.open_web_library.web.infrastructure.library;

import java.util.List;
import org.springframework.stereotype.Repository;
import com.github.dkurata38.open_web_library.library_domain.AddedLibraryItem;
import com.github.dkurata38.open_web_library.library_domain.AddedLibraryItemId;
import com.github.dkurata38.open_web_library.library_domain.AddedLibraryItemRepository;
import com.github.dkurata38.open_web_library.library_domain.OwnerId;

@Repository
public class AddedLibraryItemRepositoryImpl implements AddedLibraryItemRepository {
	private final AddedLibraryItemMapper addedLibraryItemMapper;

	public AddedLibraryItemRepositoryImpl(AddedLibraryItemMapper addedLibraryItemMapper) {
		this.addedLibraryItemMapper = addedLibraryItemMapper;
	}

	@Override
	public AddedLibraryItem findById(AddedLibraryItemId id) {
		return addedLibraryItemMapper.selectById(id);
	}

	@Override
	public List<AddedLibraryItem> findByOwnerId(OwnerId ownerId) {
		return addedLibraryItemMapper.selectAllBy(ownerId);
	}

	@Override
	public void save(AddedLibraryItem addedLibraryItem) {
		addedLibraryItemMapper.insert(addedLibraryItem);
	}
}

File: library-infra/src/main/kotlin/com/github/dkurata38/open_web_library/library_infra/AddedLibraryItemRepositoryImpl.kt
package com.github.dkurata38.open_web_library.library_infra

import com.github.dkurata38.open_web_library.library_domain.AddedLibraryItem
import com.github.dkurata38.open_web_library.library_domain.AddedLibraryItemId
import com.github.dkurata38.open_web_library.library_domain.AddedLibraryItemRepository
import com.github.dkurata38.open_web_library.library_domain.OwnerId

class AddedLibraryItemRepositoryImpl: AddedLibraryItemRepository {
	private val values: MutableList<AddedLibraryItem> = mutableListOf()
	override fun findById(id: AddedLibraryItemId): AddedLibraryItem? {
		return values
				.find { e -> e.id == id }
	}

	override fun findByOwnerId(ownerId: OwnerId): List<AddedLibraryItem> {
		return values
				.filter { e -> e.ownerId == ownerId }
	}

	override fun save(addedLibraryItem: AddedLibraryItem) {
		values.add(addedLibraryItem)
	}
}
